dotCMS/src/main/java/com/dotcms/api/system/event/local/LocalSystemEventsAPIFactory.java

```diff
@@ -0,0 +1,45 @@
+package com.dotcms.api.system.event.local;
+
+import java.io.Serializable;
+
+
+/**
+ * This singleton class provides access to the {@link LocalSystemEventsAPI} class.
+ * 
+ * @author jsanca
+ */
+@SuppressWarnings("serial")
+public class LocalSystemEventsAPIFactory implements Serializable {
+
+	private final LocalSystemEventsAPI localSystemEventsAPI;
+
+	/**
+	 * Private constructor for singleton creation.
+	 */
+	private LocalSystemEventsAPIFactory() {
+
+		this.localSystemEventsAPI = new LocalSystemEventsAPIImpl();
+	}
+
+	/**
+	 * Singleton holder using initialization on demand
+	 */
+	private static class SingletonHolder {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add a private constructor to hide the implicit public one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1118)